packages/client/lib/miner/pendingBlock.ts

```diff
@@ -21,6 +21,8 @@ import type { WithdrawalData } from '@ethereumjs/util'
 import type { TxReceipt, VM } from '@ethereumjs/vm'
 import type { BlockBuilder } from '@ethereumjs/vm/dist/buildBlock'
 
+const MAX_BLOBS_PER_BLOCK = 4
```

Comment:
I believe this would actually be hard-fork dependent, as future hardforks will drastically raise this limit (for actual sharding)
so should ideally go in the EIP params under `sharding`
but i want to handle that when it happens and not necessarily complicate the implementation now

let me know your further thoughts on this

Reply:
I would honestly argue this constant should go into common, we do this with all EIPs which define constants. See for instance EIP 1559 or also 4844 itself. If at some point `targetDataGasPerBlock` changes (or whatever constant) we can quickly spin up a new EIP and change it everywhere it is used. Since if we indeed go in the future to sharding we can anticipate on this and thus put this in common.

I would also thus argue that these checks should be Block (I think EL can announce new blocks, what if we have a malicious peer which sends us blocks with 5 blobs?). Also, maybe in other places too (buildBlock VM maybe?). So then this constant is used at multiple places and we should read it from common.

Reply:
makes sense, will change how we access and use these values from common/eip params

Reply:
@jochem-brouwer there is talk of keeping it a static high value feel free to join discussion here: https://github.com/ethereum/execution-apis/pull/404#issuecomment-1521152021